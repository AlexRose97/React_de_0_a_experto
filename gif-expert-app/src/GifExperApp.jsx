import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExperApp = () => {
  const [categories, setCategories] = useState(["Overlord", "Dragon Ball"]);
  const onAddCategoy = (newCategory) => {
    //agregar valores al array existente
    if (!categories.includes(newCategory)) {
      setCategories([newCategory, ...categories]);
    }
  };

  return (
    <>
      {/**7FyFGat9oJjnIQ3vVN9g6fFAUd95Wn4V*/}
      {/**Titulo */}
      <h1>GifExpertApp</h1>
      {/**Input */}
      <AddCategory onNewValue={(value) => onAddCategoy(value)} />
      {/**Listado_Gif*/}
      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
      {/**Item_Listado */}
    </>
  );
};
