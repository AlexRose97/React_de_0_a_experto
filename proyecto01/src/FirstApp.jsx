import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

//Definir el tipo de valor esperado en los props
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

//Definir valores default
FirstApp.defaultProps = {
  subTitle: "NO HAY SUBTITULO",
};
