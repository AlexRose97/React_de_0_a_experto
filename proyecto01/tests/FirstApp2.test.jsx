import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp/>", () => {
  const title = "este es un titulo";
  const subTitle = "este es un subtitulo";

  test("debe hacer match cin ek snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar el mensaje titulo", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle)).toHaveLength(2);
  });
});
