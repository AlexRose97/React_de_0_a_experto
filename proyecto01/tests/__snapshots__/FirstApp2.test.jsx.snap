// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe hacer match cin ek snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    este es un titulo
  </h1>
  <p>
    NO HAY SUBTITULO
  </p>
  <p>
    NO HAY SUBTITULO
  </p>
  <p />
</div>
`;
